nyct_gtfs/trip.py

```diff
@@ -275,7 +276,10 @@ def direction(self):
         Returns either "N" for northbound trains (and Grand Central bound shuttles) or "S" for southbound trains
         (and Times Square bound shuttles)
         """
-        return self.shape_id.split('..')[1][0] if '..' in self.trip_id else self.shape_id.split('.')[1][0]
+        parts = self.shape_id.split('.')
```

Comment:
```suggestion
        if not self.shape_id:
            return None
        parts = self.shape_id.split('.')
```

Reply:
Just to cover the new edge case introduced by the changes in `self.shape_id` above